### PR TITLE
Streams: takeUntil 

### PR DESCRIPTION
**Pull request by rjdestigter**
Implementation of _takeUntil_ for streams:

ToDo:
- [x] Update documentation
- [x] Tests

### CODE REVIEW

**Comment by rjdestigter** [comment]
I'm working on tests but I'm having trouble getting the current tests to run:

image

**Comment by mikearnaldi** [comment]
@rjdestigter you need to build first

**Comment by rjdestigter** [comment]
What are the build and test commands @mikearnaldi ?

**Comment by mikearnaldi** [comment]
- lerna run build (build)
- yarn jest (test)

**Comment by mikearnaldi** [comment]
they are quite long to run if you want to scope it only to effect package you can:
- yarn jest packages/effect

File: packages/effect/src/stream/index.ts [hunk 1]
```diff
@@ -444,6 +444,10 @@ function filter_<R, E, A>(stream: Stream<R, E, A>, f: F.Predicate<A>): Stream<R,
   );
 }
 
+export function filter<A, B extends A>(
```
Comment (rjdestigter) [inline packages/effect/src/stream/index.ts:447]: I also added a definition to _filter_ allowing you to refine the type of output a stream produces. To see an example, it is used on line `919` of this file where a stream of `Wrapped ` values is refined to a stream of `WrappedStream ` values.

Comment (mikearnaldi) [reply packages/effect/src/stream/index.ts:447]: This is nice, there was a variant in here:

https://github.com/Matechs-Garage/matechs-effect/blob/master/packages/effect/src/stream/index.ts#L458

But this overloading is better

**Pull request merged**